truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java

```diff
@@ -1296,13 +1296,17 @@ public Object deleteAtNullOrEmpty(RubyArray array, int index) {
         private final BranchProfile nextProfile = BranchProfile.create();
         private final BranchProfile redoProfile = BranchProfile.create();
 
+        private final RubySymbol eachSymbol;
+
         public EachNode(RubyContext context, SourceSection sourceSection) {
             super(context, sourceSection);
+            eachSymbol = getContext().getSymbolTable().getSymbol("each");
```

Comment:
It's perhaps a bit misleadingly named.  It just does `return symbolTable.getSymbol(name, ASCIIEncoding.INSTANCE);`, which will return a cached `Symbol` if it already exists or creates a new one and populates the cache otherwise.